src/impl_/pymethods.rs

```diff
@@ -1,6 +1,7 @@
-use crate::internal_tricks::{extract_cstr_or_leak_cstring, NulByteInString};
```

Comment:
Does `extract_cstr_or_leak_cstring` have other call sites? Would they benefit from the same treatment?

Reply:
Actually `extract_cstr_or_leak_cstring` is removed completely in this PR, the replacement `extract_c_string` in `pymethods.rs` is private to that module.

Reply:
Sorry for the noise. This gets dropped later on...